rm-datasource/src/main/java/io/seata/rm/datasource/exec/MulitExecutor.java

```diff
@@ -0,0 +1,190 @@
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.seata.rm.datasource.exec;
+
+import io.seata.rm.datasource.ConnectionProxy;
+import io.seata.rm.datasource.StatementProxy;
+import io.seata.rm.datasource.sql.SQLRecognizer;
+import io.seata.rm.datasource.sql.SQLType;
+import io.seata.rm.datasource.sql.struct.Row;
+import io.seata.rm.datasource.sql.struct.TableRecords;
+import io.seata.rm.datasource.undo.SQLUndoLog;
+
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Collection;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * The type MulitSql executor.
+ *  ex.
+ *  <pre>
+ *  jdbcTemplate.update("update account_tbl set money = money - ? where user_id = ?;delete from account_tbl where user_id = ?;update account_tbl set money = money - ? where user_id = ?", new Object[] {money, userId,"U10000",money,"U1000"});
+ *  </pre>
+ * @author wangwei.ying
+ *
+ * @param <T> the type parameter
+ * @param <S> the type parameter
+ */
+public class MulitExecutor<T, S extends Statement> extends AbstractDMLBaseExecutor<T, S> {
+
+
+    private List<TableRecords> updateBeforeImages = new ArrayList<>();
+    private List<TableRecords> deleteBeforeImages = new ArrayList<>();
+
+    private List<TableRecords> updateAfterImages = new ArrayList<>();
+    private List<TableRecords> deleteAfterImages = new ArrayList<>();
+    private Map<SQLRecognizer, TableRecords> tableRecordsMap = new HashMap<>(4);
+
+    /**
+     * Instantiates a new Abstract dml base executor.
+     *
+     * @param statementProxy    the statement proxy
+     * @param statementCallback the statement callback
+     * @param sqlRecognizer     the sql recognizer
+     */
+    public MulitExecutor(StatementProxy<S> statementProxy, StatementCallback<T, S> statementCallback, List<SQLRecognizer> sqlRecognizers) {
+        super(statementProxy, statementCallback, sqlRecognizers);
+    }
+
+    /**
+     * Before image table records.  only support update or deleted
+     *
+     * @return the table records
+     * @throws SQLException the sql exception
+     * @see io.seata.rm.datasource.sql.SQLVisitorFactory#getMulti(String, String) validate sqlType
+     */
+    @Override
+    protected TableRecords beforeImage() throws SQLException {
+        for (SQLRecognizer recognizer : sqlRecognizers) {
+            AbstractDMLBaseExecutor<T, S> executor = null;
+            switch (recognizer.getSQLType()) {
+                case UPDATE:
```

Comment:
1.Mulit-->Multi
2.The performance drops too badly
We can subdivide the scene:
a.Multiple tables and multiple types
b.same tale and multiple types
c.same table and same type (hope to be able to assemble the beforeImages with one sql)

Reply:
Thanks,I will subdivide it